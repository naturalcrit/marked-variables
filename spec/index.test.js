import { marked as Markdown} from 'marked';
import { markedVariables, setMarkedVariablePage }  from 'marked-variables';
import dedent from 'dedent-tabs';

// Adding `.failing()` method to `describe` or `it` will make failing tests "pass" as long as they continue to fail.
// Remove the `.failing()` method once you have fixed the issue.

// Marked.js adds line returns after closing tags on some default tokens.
// This removes those line returns for comparison sake.
String.prototype.trimReturns = function(){
	return this.replace(/\r?\n|\r/g, '').trim();
};

renderAllPages = function(pages){
	const outputs = [];
	pages.forEach((page, index)=>{
		setMarkedVariablePage(index);
		const output = Markdown(page);
		outputs.push(output);
	});

	return outputs;
};

beforeEach(() => {
	Markdown.setOptions(Markdown.getDefaults());
	Markdown.use(markedVariables());
});

describe('Block-level variables', ()=>{
	it('Handles variable assignment and recall with simple text', function() {
		const source = dedent`
			[var]: string

			$[var]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handles variable assignment and recall with multiline string', function() {
		const source = dedent`
			[var]: string
			across multiple
			lines

			$[var]`;
		const rendered = Markdown(source).replace(/\s/g, ' ').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handles variable assignment and recall with tables', function() {
		const source = dedent`
			[var]:
			##### Title
			| H1 | H2 | 
			|:---|:--:|
			| A  | B  |
			| C  | D  |
			
			$[var]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Hoists undefined variables', function() {
		const source = dedent`
			$[var]

			[var]: string`;
		const rendered = Markdown(source).replace(/\s/g, ' ').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Hoists last instance of variable', function() {
		const source = dedent`
			$[var]

			[var]: string

			[var]: new string`;
		const rendered = Markdown(source).replace(/\s/g, ' ').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handles complex hoisting', function() {
		const source = dedent`
			$[titleAndName]: $[title] $[fullName]

			$[title]: Mr.

			$[fullName]: $[firstName] $[lastName]

			[firstName]: Bob

			Welcome, $[titleAndName]!

			[lastName]: Jacob

			[lastName]: $[lastName]son`;
		const rendered = Markdown(source).replace(/\s/g, ' ').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Displays unparsed variable if not defined : block', function() {
		const source = dedent`$[var]`;
		const rendered = Markdown(source).replace(/\s/g, ' ').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Displays unparsed variable if not defined : inline', function() {
		const source = dedent`Hello $[var]`;
		const rendered = Markdown(source).replace(/\s/g, ' ').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handles variable reassignment', function() {
		const source = dedent`
			[var]: one

			$[var]

			[var]: two

			$[var]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handles variable reassignment with hoisting', function() {
		const source = dedent`
			$[var]
		
			[var]: one

			$[var]

			[var]: two

			$[var]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Ignores undefined variables that can\'t be hoisted', function() {
		const source = dedent`
			$[var](My name is $[first] $[last])

			$[last]: Jones`;
		const rendered = Markdown(source).replace(/\s/g, ' ').trimReturns();
		expect(rendered).toMatchSnapshot();
	});
});

describe('Inline-level variables', ()=>{
	it('Handles variable assignment and recall with simple text', function() {
		const source = dedent`
			$[var](string)

			$[var]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Hoists undefined variables when possible', function() {
		const source = dedent`
			$[var](My name is $[name] Jones)

			[name]: Bob`;
		const rendered = Markdown(source).replace(/\s/g, ' ').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Hoists last instance of variable', function() {
		const source = dedent`
			$[var](My name is $[name] Jones)

			$[name](Bob)

			[name]: Bill`;
		const rendered = Markdown(source).replace(/\s/g, ' ').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Only captures nested parens if balanced', function() {
		const source = dedent`
			$[var1](A variable (with nested parens) inside)

			$[var1]

			$[var2](A variable ) with unbalanced parens)

			$[var2]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});
});

describe('Math', ()=>{
	it('Handles simple math using numbers only', function() {
		const source = dedent`
			$[1 + 3 * 5 - (1 / 4)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handles round function', function() {
		const source = dedent`
			$[round(1/4)]`;
		const rendered = Markdown(source).replace(/\s/g, ' ').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handles floor function', function() {
		const source = dedent`
			$[floor(0.6)]`;
		const rendered = Markdown(source).replace(/\s/g, ' ').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handles ceil function', function() {
		const source = dedent`
			$[ceil(0.2)]`;
		const rendered = Markdown(source).replace(/\s/g, ' ').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handles nested functions', function() {
		const source = dedent`
			$[ceil(floor(round(0.6)))]`;
		const rendered = Markdown(source).replace(/\s/g, ' ').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handles simple math with variables', function() {
		const source = dedent`
			$[num1]: 5

			$[num2]: 4

			Answer is $[answer]($[1 + 3 * num1 - (1 / num2)]).`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handles variable incrementing', function() {
		const source = dedent`
			$[num1]: 5

			Increment num1 to get $[num1]($[num1 + 1]) and again to $[num1]($[num1 + 1]).`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});
});

describe('Code blocks', ()=>{
	it('Ignores all variables in fenced code blocks', function() {
		const source = dedent`
			\`\`\`
			[var]: string

			$[var]

			$[var](new string)
			\`\`\``;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Ignores all variables in indented code blocks', function() {
		const source = dedent`
			test

			    [var]: string

			    $[var]

			    $[var](new string)`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Ignores all variables in inline code blocks', function() {
		const source = '[var](Hello) `[link](url)`. This `[var] does not work`';
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});
});

describe('Normal Links and Images', ()=>{
	it('Renders normal images', function() {
		const source = `![alt text](url)`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Renders normal images with a title', function() {
		const source = 'An image ![alt text](url "and title")!';
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Applies curly injectors to images', function() {
		const source = `![alt text](url){width:100px}`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Renders normal links', function() {
		const source = 'A Link to my [website](url)!';
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Renders normal links with a title', function() {
		const source = 'A Link to my [website](url "and title")!';
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});
});

describe('Cross-page variables', ()=>{
	it('Handles variable assignment and recall across pages', function() {
		const source0 = `[var]: string`;
		const source1 = `$[var]`;
		const rendered = renderAllPages([source0, source1]).join('\n\\page\n').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handles hoisting across pages', function() {
		const source0 = `$[var]`;
		const source1 = `[var]: string`;
		renderAllPages([source0, source1]).join('\n\\page\n').trimReturns();	//Requires one full render of document before hoisting is picked up
		const rendered = renderAllPages([source0, source1]).join('\n\\page\n').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handles reassignment and hoisting across pages', function() {
		const source0 = `$[var]\n\n[var]: one\n\n$[var]`;
		const source1 = `[var]: two\n\n$[var]`;
		renderAllPages([source0, source1]).join('\n\\page\n').trimReturns();	//Requires one full render of document before hoisting is picked up
		const rendered = renderAllPages([source0, source1]).join('\n\\page\n').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Page numbering across pages : default', function() {
		const source0 = `$[HB_pageNumber]\n\n`;
		const source1 = `$[HB_pageNumber]\n\n`;
		renderAllPages([source0, source1]).join('\n\\page\n').trimReturns();	//Requires one full render of document before hoisting is picked up
		const rendered = renderAllPages([source0, source1]).join('\n\\page\n').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Page numbering across pages : custom page number (Number)', function() {
		const source0 = `[HB_pageNumber]:100\n\n$[HB_pageNumber]\n\n`;
		const source1 = `$[HB_pageNumber]\n\n`;
		renderAllPages([source0, source1]).join('\n\\page\n').trimReturns();	//Requires one full render of document before hoisting is picked up
		const rendered = renderAllPages([source0, source1]).join('\n\\page\n').trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Page numbering across pages : custom page number (NaN)', function() {
		const source0 = `[HB_pageNumber]:a\n\n$[HB_pageNumber]\n\n`;
		const source1 = `$[HB_pageNumber]\n\n`;
		renderAllPages([source0, source1]).join('\n\\page\n').trimReturns();	//Requires one full render of document before hoisting is picked up
		const rendered = renderAllPages([source0, source1]).join('\n\\page\n').trimReturns();
		expect(rendered).toMatchSnapshot();
	});
});

describe('Math function parameter handling', ()=>{
	it('allows variables in single-parameter functions', function() {
		const source = '[var]:4.1\n\n$[floor(var)]';
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});
	it('allows one variable and a number in two-parameter functions', function() {
		const source = '[var]:4\n\n$[min(1,var)]';
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});
	it('allows two variables in two-parameter functions', function() {
		const source = '[var1]:4\n\n[var2]:8\n\n$[min(var1,var2)]';
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});
});

describe('Variable names that are subsets of other names', ()=>{
	it('do not conflict with function names', function() {
		const source = `[a]: -1\n\n$[abs(a)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('do not conflict with other variable names', function() {
		const source = `[ab]: 2\n\n[aba]: 8\n\n[ba]: 4\n\n$[ab + aba + ba]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});
});

describe('Regression Tests', ()=>{
	it('Don\'t Eat all the parentheticals!', function() {
		const source='\n|  title 1  | title 2 | title 3 | title 4|\n|-----------|---------|---------|--------|\n|[foo](bar) |  Ipsum  |    )    |   )    |\n';
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handle Extra spaces in image alt-text 1', function(){
		const source='![ where is my image??](http://i.imgur.com/hMna6G0.png)';
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handle Extra spaces in image alt-text 2', function(){
		const source='![where  is my image??](http://i.imgur.com/hMna6G0.png)';
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handle Extra spaces in image alt-text 3', function(){
		const source='![where is my image?? ](http://i.imgur.com/hMna6G0.png)';
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Handle Extra spaces in image alt-text 4', function(){
		const source='![where is my image??](http://i.imgur.com/hMna6G0.png){height=20%,width=20%}';
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});
});

describe('Custom Math Function Tests', ()=>{
	it('Sign Test', function() {
		const source = `[a]: 13\n\n[b]: -11\n\nPositive: $[sign(a)]\n\nNegative: $[sign(b)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Signed Test', function() {
		const source = `[a]: 13\n\n[b]: -11\n\nPositive: $[signed(a)]\n\nNegative: $[signed(b)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Roman Numerals Test', function() {
		const source = `[a]: 18\n\nRoman Numeral: $[toRomans(a)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Roman Numerals Test - Uppercase', function() {
		const source = `[a]: 18\n\nRoman Numeral: $[toRomansUpper(a)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Roman Numerals Test - Lowercase', function() {
		const source = `[a]: 18\n\nRoman Numeral: $[toRomansLower(a)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Number to Characters Test', function() {
		const source = `[a]: 18\n\n[b]: 39\n\nCharacters: $[toChar(a)] $[toChar(b)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Number to Characters Test - Uppercase', function() {
		const source = `[a]: 18\n\n[b]: 39\n\nCharacters: $[toCharUpper(a)] $[toCharUpper(b)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Number to Characters Test - Lowercase', function() {
		const source = `[a]: 18\n\n[b]: 39\n\nCharacters: $[toCharLower(a)] $[toCharLower(b)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Number to Words Test', function() {
		const source = `[a]: 80085\n\nWords: $[toWords(a)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Number to Words Test - Uppercase', function() {
		const source = `[a]: 80085\n\nWords: $[toWordsUpper(a)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Number to Words Test - Lowercase', function() {
		const source = `[a]: 80085\n\nWords: $[toWordsLower(a)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});

	it('Number to Words Test - Capitalized', function() {
		const source = `[a]: 80085\n\nWords: $[toWordsCaps(a)]`;
		const rendered = Markdown(source).trimReturns();
		expect(rendered).toMatchSnapshot();
	});
});
