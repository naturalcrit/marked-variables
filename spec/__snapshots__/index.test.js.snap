// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Block-level variables Displays unparsed variable if not defined : block 1`] = `"<p>$[var]</p>"`;

exports[`Block-level variables Displays unparsed variable if not defined : inline 1`] = `"<p>Hello $[var]</p>"`;

exports[`Block-level variables Handles complex hoisting 1`] = `"<p>Welcome, Mr. Bob Jacobson!</p>"`;

exports[`Block-level variables Handles variable assignment and recall with multiline string 1`] = `"<p>string across multiple lines</p>"`;

exports[`Block-level variables Handles variable assignment and recall with simple text 1`] = `"<p>string</p>"`;

exports[`Block-level variables Handles variable assignment and recall with tables 1`] = `"<h5>Title</h5><table><thead><tr><th align="left">H1</th><th align="center">H2</th></tr></thead><tbody><tr><td align="left">A</td><td align="center">B</td></tr><tr><td align="left">C</td><td align="center">D</td></tr></tbody></table>"`;

exports[`Block-level variables Handles variable reassignment 1`] = `"<p>one</p><p>two</p>"`;

exports[`Block-level variables Handles variable reassignment with hoisting 1`] = `"<p>two</p><p>one</p><p>two</p>"`;

exports[`Block-level variables Hoists last instance of variable 1`] = `"<p>new string</p>"`;

exports[`Block-level variables Hoists undefined variables 1`] = `"<p>string</p>"`;

exports[`Block-level variables Ignores undefined variables that can't be hoisted 1`] = `"<p>My name is $[first] Jones</p>"`;

exports[`Code blocks Ignores all variables in fenced code blocks 1`] = `"<pre><code>[var]: string$[var]$[var](new string)</code></pre>"`;

exports[`Code blocks Ignores all variables in indented code blocks 1`] = `"<p>test</p><pre><code>[var]: string$[var]$[var](new string)</code></pre>"`;

exports[`Code blocks Ignores all variables in inline code blocks 1`] = `"<p><a href="Hello">var</a> <code>[link](url)</code>. This <code>[var] does not work</code></p>"`;

exports[`Cross-page variables Handles hoisting across pages 1`] = `"<p>string</p>\\page"`;

exports[`Cross-page variables Handles reassignment and hoisting across pages 1`] = `"<p>two</p><p>one</p>\\page<p>two</p>"`;

exports[`Cross-page variables Handles variable assignment and recall across pages 1`] = `"\\page<p>string</p>"`;

exports[`Cross-page variables Page numbering across pages : custom page number (NaN) 1`] = `"<p>a</p>\\page<p>a</p>"`;

exports[`Cross-page variables Page numbering across pages : custom page number (Number) 1`] = `"<p>100</p>\\page<p>100</p>"`;

exports[`Cross-page variables Page numbering across pages : default 1`] = `"<p>$[HB_pageNumber]</p>\\page<p>$[HB_pageNumber]</p>"`;

exports[`Custom Math Function Tests Number to Characters Test - Lowercase 1`] = `"<p>Characters: r am</p>"`;

exports[`Custom Math Function Tests Number to Characters Test - Uppercase 1`] = `"<p>Characters: R AM</p>"`;

exports[`Custom Math Function Tests Number to Characters Test 1`] = `"<p>Characters: R AM</p>"`;

exports[`Custom Math Function Tests Number to Words Test - Capitalized 1`] = `"<p>Words: Eighty Thousand And Eighty-Five</p>"`;

exports[`Custom Math Function Tests Number to Words Test - Lowercase 1`] = `"<p>Words: eighty thousand and eighty-five</p>"`;

exports[`Custom Math Function Tests Number to Words Test - Uppercase 1`] = `"<p>Words: EIGHTY THOUSAND AND EIGHTY-FIVE</p>"`;

exports[`Custom Math Function Tests Number to Words Test 1`] = `"<p>Words: eighty thousand and eighty-five</p>"`;

exports[`Custom Math Function Tests Roman Numerals Test - Lowercase 1`] = `"<p>Roman Numeral: xviii</p>"`;

exports[`Custom Math Function Tests Roman Numerals Test - Uppercase 1`] = `"<p>Roman Numeral: XVIII</p>"`;

exports[`Custom Math Function Tests Roman Numerals Test 1`] = `"<p>Roman Numeral: XVIII</p>"`;

exports[`Custom Math Function Tests Sign Test 1`] = `"<p>Positive: +</p><p>Negative: -</p>"`;

exports[`Custom Math Function Tests Signed Test 1`] = `"<p>Positive: +13</p><p>Negative: -11</p>"`;

exports[`Inline-level variables Handles variable assignment and recall with simple text 1`] = `"<p>string</p><p>string</p>"`;

exports[`Inline-level variables Hoists last instance of variable 1`] = `"<p>My name is Bill Jones</p> <p>Bob</p>"`;

exports[`Inline-level variables Hoists undefined variables when possible 1`] = `"<p>My name is Bob Jones</p>"`;

exports[`Inline-level variables Only captures nested parens if balanced 1`] = `"<p>A variable (with nested parens) inside</p><p>A variable (with nested parens) inside</p><p>A variable with unbalanced parens)</p><p>A variable</p>"`;

exports[`Math Handles ceil function 1`] = `"<p>1</p>"`;

exports[`Math Handles floor function 1`] = `"<p>0</p>"`;

exports[`Math Handles nested functions 1`] = `"<p>1</p>"`;

exports[`Math Handles round function 1`] = `"<p>0</p>"`;

exports[`Math Handles simple math using numbers only 1`] = `"<p>15.75</p>"`;

exports[`Math Handles simple math with variables 1`] = `"<p>Answer is 15.75.</p>"`;

exports[`Math Handles variable incrementing 1`] = `"<p>Increment num1 to get 6 and again to 7.</p>"`;

exports[`Math function parameter handling allows one variable and a number in two-parameter functions 1`] = `"<p>1</p>"`;

exports[`Math function parameter handling allows two variables in two-parameter functions 1`] = `"<p>4</p>"`;

exports[`Math function parameter handling allows variables in single-parameter functions 1`] = `"<p>4</p>"`;

exports[`Normal Links and Images Applies curly injectors to images 1`] = `"<p><img src="url" alt="alt text">{width:100px}</p>"`;

exports[`Normal Links and Images Renders normal images 1`] = `"<p><img src="url" alt="alt text"></p>"`;

exports[`Normal Links and Images Renders normal images with a title 1`] = `"<p>An image <img src="url" alt="alt text" title="and title">!</p>"`;

exports[`Normal Links and Images Renders normal links 1`] = `"<p>A Link to my <a href="url">website</a>!</p>"`;

exports[`Normal Links and Images Renders normal links with a title 1`] = `"<p>A Link to my <a href="url" title="and title">website</a>!</p>"`;

exports[`Regression Tests Don't Eat all the parentheticals! 1`] = `"<table><thead><tr><th>title 1</th><th>title 2</th><th>title 3</th><th>title 4</th></tr></thead><tbody><tr><td><a href="bar">foo</a></td><td>Ipsum</td><td>)</td><td>)</td></tr></tbody></table>"`;

exports[`Regression Tests Handle Extra spaces in image alt-text 1 1`] = `"<p><img src="http://i.imgur.com/hMna6G0.png" alt="where is my image??"></p>"`;

exports[`Regression Tests Handle Extra spaces in image alt-text 2 1`] = `"<p><img src="http://i.imgur.com/hMna6G0.png" alt="where is my image??"></p>"`;

exports[`Regression Tests Handle Extra spaces in image alt-text 3 1`] = `"<p><img src="http://i.imgur.com/hMna6G0.png" alt="where is my image??"></p>"`;

exports[`Regression Tests Handle Extra spaces in image alt-text 4 1`] = `"<p><img src="http://i.imgur.com/hMna6G0.png" alt="where is my image??">{height=20%,width=20%}</p>"`;

exports[`Variable names that are subsets of other names do not conflict with function names 1`] = `"<p>1</p>"`;

exports[`Variable names that are subsets of other names do not conflict with other variable names 1`] = `"<p>14</p>"`;
